test/unit/module/core/simd_cast.cpp

```diff
@@ -117,30 +129,71 @@ simd_cast_logical_test()
   simd_cast_test<eve::logical<T>, eve::logical<U>>();
 }
 
+template<typename Wide>
+consteval bool
+rvv_check_enable()
+{
+  using scalar_type = typename Wide::value_type;
+  return (eve::fundamental_cardinal_v<scalar_type> == eve::cardinal_v<Wide>
+          || eve::expected_cardinal_v<scalar_type> == eve::cardinal_v<Wide>);
+}
+
 TTS_CASE_TPL("simd_cast, plain", eve::test::simd::all_types)
 <typename T>(tts::type<T>)
 {
-  eve::detail::for_<0, 1, std::bit_width((std::size_t)T::size() * 2)>(
-      []<typename Idx>(Idx)
-      {
-        simd_cast_test<T, eve::wide<std::uint8_t, eve::fixed<1 << Idx::value>>>();
-        simd_cast_test<T, eve::wide<std::uint16_t, eve::fixed<1 << Idx::value>>>();
-        simd_cast_test<T, eve::wide<std::uint32_t, eve::fixed<1 << Idx::value>>>();
-        simd_cast_test<T, eve::wide<std::uint64_t, eve::fixed<1 << Idx::value>>>();
-      });
+  // for RISC-V we have too big memory consumption (for vlen=128):
+  // as input we can have wide<std::int8_t, 128>,
+  // so common code will try to create wide<std::uint64_t, 128>, but on physical registers we can
+  // create wide<std::uint64_t, 16>.
+  // Other cases will have the same problem, so greatly reduce testing for RISC-V
```

Comment:
We should test wide<std::uint64_t, 32>, not even 16.
We should test up to max len * 2. 

If we don't want to test those - this is not the right place to do it.